src/ipc/process.cpp

```diff
@@ -54,7 +60,100 @@ class ProcessImpl : public Process
         }
         return true;
     }
+    int connect(const fs::path& data_dir,
+                const std::string& dest_exe_name,
+                std::string& address,
+                std::string& error) override;
+    int bind(const fs::path& data_dir, const std::string& exe_name, std::string& address, std::string& error) override;
 };
+
+static bool ParseAddress(std::string& address,
```

Comment:
52917ad4be1ca574543e854fdc154e88d65ce57f: this could use a test.

Or maybe we can introduce something akin to `CNetAddr` for socket addresses? Fortunately this input can only be provided by the user, so we don't have to be overzealous about sanitising.